src/cryptonote_basic/hardfork.cpp

```diff
@@ -393,7 +393,7 @@ uint8_t HardFork::get_next_version() const
 {
   CRITICAL_REGION_LOCAL(lock);
   uint64_t height = db.height();
-  for (unsigned int n = heights.size() - 1; n > 0; --n) {
+  for (int n = heights.size() - 1; n >= 0; --n) {
```

Comment:
In which case would this break and how ?

Reply:
rewriting this cycle with reverse iterators will make it more clear

Reply:
@moneromooo-monero 

With the previous code, it wouldn't reach the `n = 0` case, and hence fails at L669
```c++
    ASSERT_EQ(hf.get_earliest_ideal_height_for_version(1), 0);
```
with the error message being:
```
tests/unit_tests/hardfork.cpp:669: Failure
Value of: 0
Expected: hf.get_earliest_ideal_height_for_version(1)
Which is: 2
[  FAILED  ] get.earliest_ideal_height (0 ms)
```

@naughtyfox 
Agreed, thanks for suggesting!